### PR TITLE
Add object apis

### PR DESCRIPTION
**Pull request by yongpingchen**
- add PostObject
- add GetObject
- add DeleteObject
- add DeleteBucket

### CODE REVIEW

**Review by satoshikumano — COMMENTED** [review]
- Scope definition in author.go is the best location?
- Have you considered defining types for each scope instead of defining enum?

example:
```golang
type Scope interface {
	Path() string
}

type AppScope struct {
    AppID string
}

func (ac AppScope) Path() string {
	return "apps/" + ac.AppID
}
```

**Comment by satoshikumano** [comment]
> CreateObject

There's some variation of POST/ PUT/ PATCH operaion of object.
Whether we implement all of them or not in this PR, It would be worthy of considering the shape each of APIs.
(In Kii Cloud JS SDK, mapping of those REST API operation to SDK API is very hard to understand 😅)

POST :
- create new object

PUT : 
- create new object with the specified ID (If-None-Match: *) (fail if already exist.)
- update object (If-Match : not present) ( create new if not exist.)
- update object (If-Match : present)

PATCH:
- update object (If-Match : not present)
- update object (If-Match : present)

**Comment by yongpingchen** [comment]
@satoshikumano 

> Scope definition in author.go is the best location?

I will put them on `api.go`

> Have you considered defining types for each scope instead of defining enum?

em, seems to use interface clearer. Like this: 

```go
type Scope interface {
	Path() string
}

type AppScope struct {
}

func (as AppScope) Path() string {
	return ""
}

type UserScope struct {
	ID string
}

func (us UserScope) Path() string {
	return "users/" + us.ID + "/"
}

```

> There's some variation of POST/ PUT/ PATCH operaion of object.
Whether we implement all of them or not in this PR, It would be worthy of considering the shape each of APIs.

So many pattern😅. I would like to have the basic APIs in this PR. To make it external in the future, better to rename `CreateObject` with `PostNewObject`.

**Comment by yongpingchen** [comment]
@satoshikumano fixed the implementation and add tests. Please confirm

File: bucket.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// GroupBucket represents group bucket
+type GroupBucket struct {
+	BucketName string
+	GroupID    string
+}
+
+// Path generates path string of group bucket
+func (ub GroupBucket) Path() string {
```
Comment (satoshikumano) [inline bucket.go]: gb

File: bucket.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// ThingBucket represents thing bucket
+type ThingBucket struct {
+	BucketName string
+	ThingID    string
+}
+
+// Path generates path string of user bucket
+func (ub ThingBucket) Path() string {
```
Comment (satoshikumano) [inline bucket.go]: tb

File: author.go [hunk 3]
```diff
@@ -401,25 +401,11 @@ func (a *APIAuthor) ListEndNodes(gatewayID string, listPara ListRequest) (*ListE
 
 // CreateThingScopeObject create Thing scope object
 func (a APIAuthor) CreateThingScopeObject(thingID, bucketName string, object map[string]interface{}) (*CreateObjectResponse, error) {
```
Comment (satoshikumano) [inline author.go]: これは前からあったのですね。
Deprecate にした方がいいかもね。

**Review by satoshikumano — COMMENTED** [review]
Left comments.

File: author.go (hunk 3 above)
Comment (satoshikumano) [reply author.go]: https://blog.golang.org/godoc-documenting-go-code

**Comment by yongpingchen** [comment]
@satoshikumano fixed.

**Pull request merged**